lib/binary_parser.ts

```diff
@@ -68,19 +68,29 @@ interface ParserOptions {
 type Types =
```

Comment:
You could union these types together instead of redeclaring them
```
type CompltexTypes = 'String' | 'Nest' ....
type PrimitiveTypes = 'UInt32LE' | 'Int16LE'

type Types = CompltexTypes | PrimitiveTypes
```